src/main/java/no/nav/pto/veilarbportefolje/arenaaktiviteter/GruppeAktivitetRepository.java

```diff
@@ -0,0 +1,170 @@
+import no.nav.pto.veilarbportefolje.aktiviteter.AktivitetStatus;
+import no.nav.pto.veilarbportefolje.aktiviteter.AktivitetTyper;
+import no.nav.pto.veilarbportefolje.arenaaktiviteter.arenaDTO.GruppeAktivitetInnhold;
+import no.nav.pto.veilarbportefolje.arenaaktiviteter.arenaDTO.GruppeAktivitetSchedueldDTO;
+import no.nav.pto.veilarbportefolje.domene.value.PersonId;
+import no.nav.sbl.sql.SqlUtils;
+import no.nav.sbl.sql.where.WhereClause;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.Timestamp;
+import java.time.ZonedDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+import static no.nav.pto.veilarbportefolje.arenaaktiviteter.ArenaAktivitetUtils.getDateOrNull;
+import static no.nav.pto.veilarbportefolje.database.Table.GRUPPE_AKTIVITER.*;
+import static no.nav.pto.veilarbportefolje.util.DateUtils.toTimestamp;
+
+
+@Slf4j
+@Repository
+@Transactional
+@RequiredArgsConstructor
+public class GruppeAktivitetRepository {
+    private final JdbcTemplate db;
+    private final AktivitetDAO aktivitetDAO;
+
+    public void upsertGruppeAktivitet(GruppeAktivitetInnhold gruppeAktivitet, AktorId aktorId, boolean aktiv) {
+        // Fra dato kan ha verdien null, det tilsier at aktiviteten varer en hel dag
+        ZonedDateTime tilDato = getDateOrNull(gruppeAktivitet.getAktivitetperiodeTil(), true);
+        ZonedDateTime fraDato = gruppeAktivitet.getAktivitetperiodeFra() == null ? tilDato.minusDays(1) : getDateOrNull(gruppeAktivitet.getAktivitetperiodeFra());
+        String aktivChar = aktiv ? "J" : "N";
+
+        SqlUtils.upsert(db, TABLE_NAME)
+                .set(MOTEPLAN_ID, gruppeAktivitet.getMoteplanId())
+                .set(VEILEDNINGDELTAKER_ID, gruppeAktivitet.getVeiledningdeltakerId())
+                .set(AKTOERID, aktorId.get())
+                .set(MOTEPLAN_STARTDATO, toTimestamp(fraDato))
+                .set(MOTEPLAN_SLUTTDATO, toTimestamp(tilDato))
+                .set(HENDELSE_ID, gruppeAktivitet.getHendelseId())
+                .set(AKTIV, aktivChar)
+                .where(
+                        WhereClause.equals(MOTEPLAN_ID, gruppeAktivitet.getMoteplanId())
+                                .and(WhereClause.equals(VEILEDNINGDELTAKER_ID, gruppeAktivitet.getVeiledningdeltakerId()))
+                ).execute();
+    }
+
+    /**
+     * Setter aktivitet til inaktiv kun hvis aktiviteten er utgatt.
+     * Implementert til aa forhindre race condition mellom daglig jobb og kafka.
+     */
+    public int oppdaterUtgattAktivStatus(String moteplanId, String veiledningdeltakerId, AktorId aktorId, PersonId personId) {
+        String updateSql = String.format(
```

Comment:
usage of String.format, and later db.update is bit difficult to read for me. It requires some effort to connect all placeholders and their values, since they are in different places.

Reply:
is it necessary to have field: AKTIV if there is already another field which has the same meaning: MOTEPLAN_SLUTTDATO. Is it possible to have activity that is not active, with null as expiration date

Reply:
I for one do no longer trust that stuff like MOTEPLAN_SLUTTDATO being null means inactive in the case of arena tables. E.g. with "tiltak" it is possible to have all date fields set to null and it can stil be an active "tiltak".